### ISSUE TITLE
Fragmentation of neurons (i2g)

### ISSUE THREAD

**Issue by dlee138** [Statistical-Connectomics-Sp15/intro#137]
What causes fragmentation of neurons and why don't synapses have them? Is it merely the shape of the neurons that causes this? (I.e. protruding areas as shown in fig 3)

**Comment by akim1** [comment]
I think that "fragmentation" refers to the fact that the segmentation algorithm doesn't correctly delineate the boundary of the neuron all the way to the dendritic spine. The effect is that the neuron is "fragmented" and not whole in the eyes of the algorithm. The result is that there is a large "graph error" since a whole edge is missing, while visually, the error is very minimal.

**Comment by whock** [comment]
I agree that what @akim1 brought up likely contributes to this issue. Dendritic spine necks are really thin and under confocal microscopic approaches can actually be invisible (i.e. smaller than resolution of scope). For EM, the neck itself will be visible but could exceed the thinness that the algorithm uses to determine boundaries. 

Another issue may be that the synapse/spine environment can look pretty different from that of the dendritic shaft it attaches to, and so some algorithms may tag them as being separate. For example, one highly-cited algorithm called Blobworld [1] uses textural differences to set boundaries. As does the algorithm in the I2G paper (according to the text). The synapse has features like vesicles, different cytoskeletal molecules, etc, that differ from the shaft (and the shaft has stuff the synapse doesn't). So that may fool the segmentation algorithm. 

[1] http://www.cs.berkeley.edu/~malik/papers/CBGM-blobworld.pdf

**Comment by willgray** [comment]
Good thoughts here - I'll add that for I2G, we assume that synapses are "point" connections between two neurons, so we ignore topology, morphology, etc.  We just care about neurons connecting.  Agree that this is incomplete, but we felt like starting simple and getting a result was better than not having one!
So in our model, synapses can't fragment - they are there or not there (and typically form a tight cluster).  We do use the pixels to assign incident neurons...
